src/etc/inc/interfaces.inc

```diff
@@ -3272,18 +3272,23 @@ function interface_configure($interface = "wan", $reloadall = false, $linkupeven
 		interface_wireless_configure($realif, $wancfg, $wancfg['wireless']);
 	}
 
+	/* Get the vendor MAC.  Use source dependent upon whether or not booting. */
 	$current_mac = get_interface_mac($realhwif);
-	$vendor_mac = get_interface_vendor_mac($realhwif);
-	$spoof_mac = $wancfg['spoofmac'] ?: $vendor_mac;
+	if (platform_booting()) {
+		$vendor_mac = $current_mac;
+	} else {
+		$vendor_mac = get_interface_vendor_mac($realhwif);
+	}
```

Comment:
Unfortunately you cannot trust interface_configure() will run only once during boot, there are edge cases where it runs more than once and in these cases you would end up with spoofed mac being stored in vendor_mac file.

You should check if there is already a definition in vendor_mac file, and if yes, consider that definition as vendor mac